filament/src/materials/skybox.mat

```diff
@@ -58,6 +58,14 @@ fragment {
 
 vertex {
     void materialVertex(inout MaterialVertexInputs material) {
-        material.eyeDirection.xyz = material.worldPosition.xyz;
+        vec4 p = getPosition();
+        // GL convention to inverted DX convention
+        p.z = p.z * -0.5 + 0.5;
+        vec4 worldPosition = getWorldFromClipMatrix() * p;
+        // Getting the true world position would require dividing by w, but since this is a skybox
+        // at inifinity, this results in very large numbers for material.eyeDirection.
+        // Since the eyeDirection is only used as a direction vector in the fragment shader, we can
+        // skip that step to improve precision.
+        material.eyeDirection.xyz = worldPosition.xyz;
```

Comment:
I'm even wondering if we shouldn't `normalize()` here instead of in the fragment shader (when the sun disk is enabled) -- that would make it even clearer that the w-divide is not needed.

Can you add a comment that this code is taken from `computeWorldPosition` and assumes `VERTEX_DOMAIN_DEVICE`.

Reply:
We can't `normalize()` here because `eyeDirection` still needs to be interpolated. See the difference:

https://github.com/google/filament/assets/5298046/1b9a573a-aeb7-4eaf-b06f-078a277268d5